### PR TITLE
hwmv2: riscv: intel_niosv: Port to HWMv2

### PR DESCRIPTION
**Pull request by golowanow**
Move to HWMv2 and convert from `soc/riscv/intel_niosv`  SoC family `INTEL_NIOSV` series `NIOSV` SoCs `NIOSV_M` and `NIOSV_G`.

Move and convert to HWMv2 `riscv/niosv_g` board configuration.
Move and convert to HWMv2 `riscv/niosv_m` board configuration.

### CODE REVIEW

File: soc/intel/intel_niosv/niosv/Kconfig.defconfig.series
Comment (golowanow) [inline soc/intel/intel_niosv/niosv/Kconfig.defconfig.series:8]: needs this for HWMv2 build with correct `arch/riscv` directory overriding the value from here

https://github.com/zephyrproject-rtos/zephyr/blob/592a7cd6a750bdc715f07f004a25226618a89012/arch/riscv/Kconfig#L7-L10

@tejlmand is the above the expected way for HWMv2 to deal with ARCH, `riscv` in particular ?

Comment (nordicjm) [reply soc/intel/intel_niosv/niosv/Kconfig.defconfig.series:8]: Haven't reviewed this but riscv porting is not working at present due to differences in name of arch and folders, need to cherry-pick some changes over

Comment (golowanow) [reply soc/intel/intel_niosv/niosv/Kconfig.defconfig.series:8]: > Haven't reviewed this but riscv porting is not working at present due to differences in name of arch and folders, need to cherry-pick some changes over

it should be this Kconfig change to allow riscv build, at least it is for intel_niosv

Comment (nordicjm) [reply soc/intel/intel_niosv/niosv/Kconfig.defconfig.series:8]: https://github.com/zephyrproject-rtos/zephyr/pull/68853 is the fix

**Comment by golowanow** [comment]
rebased including #68853

File: soc/intel/intel_niosv/CMakeLists.txt
Comment (nordicjm) [inline soc/intel/intel_niosv/CMakeLists.txt:7]: Have moved this as per https://github.com/zephyrproject-rtos/zephyr/pull/68872 but need to wait for feedback to ensure it is acceptable (marking as NACK to prevent merge)

File: boards/intel/niosv_m/Kconfig.defconfig
Comment (nordicjm) [inline boards/intel/niosv_m/Kconfig.defconfig:3]: delete this file

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: boards/intel/niosv_m/Kconfig.defconfig
Comment (golowanow) [reply boards/intel/niosv_m/Kconfig.defconfig]: done

File: soc/intel/intel_niosv/CMakeLists.txt
Comment (golowanow) [reply soc/intel/intel_niosv/CMakeLists.txt]: changed

**Comment by golowanow** [comment]
rebased to the latest `collab-hwm` changes

**Comment by golowanow** [comment]
@nashif this SoC/board after the move has no explicit maintainers

**Comment by golowanow** [comment]
@nordicjm, @tejlmand  up to your attention

File: soc/intel/intel_niosv/niosv/Kconfig
Comment (nordicjm) [inline soc/intel/intel_niosv/niosv/Kconfig:17]: no quotes on help text

**Review by nordicjm — COMMENTED** [review]
minor nit.

File: soc/intel/intel_niosv/niosv/Kconfig
Comment (golowanow) [reply soc/intel/intel_niosv/niosv/Kconfig]: done

File: boards/intel/niosv_g/Kconfig.niosv_g
Comment (tejlmand) [inline boards/intel/niosv_g/Kconfig.niosv_g:6]: Please remove bool, same reason as given in other pr.
```suggestion
```

File: boards/intel/niosv_m/Kconfig.niosv_m
Comment (tejlmand) [inline boards/intel/niosv_m/Kconfig.niosv_m:6]: Please remove bool.
```suggestion
```

**Review by tejlmand — COMMENTED** [review]
minor nits noticed.

File: boards/intel/niosv_g/Kconfig.niosv_g
Comment (golowanow) [reply boards/intel/niosv_g/Kconfig.niosv_g]: removed

File: boards/intel/niosv_m/Kconfig.niosv_m
Comment (golowanow) [reply boards/intel/niosv_m/Kconfig.niosv_m]: removed

**Review by tejlmand — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Pull request merged**
